query/src/formats/output_format.rs

```diff
@@ -60,16 +83,21 @@ impl OutputFormatType {
 
 impl Default for OutputFormatType {
     fn default() -> Self {
-        Self::Tsv
+        Self::Tsv(HeaderConfig::new(false, false))
     }
 }
 
 impl FromStr for OutputFormatType {
     type Err = ErrorCode;
     fn from_str(s: &str) -> std::result::Result<Self, ErrorCode> {
+        // todo(youngsofun):  refactor with factory
         match s.to_uppercase().as_str() {
-            "TSV" => Ok(OutputFormatType::Tsv),
-            "CSV" => Ok(OutputFormatType::Csv),
+            "TSV" => Ok(OutputFormatType::Tsv(HeaderConfig::new(false, false))),
+            "TSVWITHNAMES" => Ok(OutputFormatType::Tsv(HeaderConfig::new(true, false))),
```

Comment:
How about new enum type? 
```
enum OutputFormatType {
    TSV,
    TSVWithName,
    TSVWithNameAndType,
    CSV,
    CSVWithName,
    CSVWithNameAndType
}
```

Reply:
good idea!

Reply:
@zhang2014 maybe it is time to use your FormatFactory,  more freedom and less code to handle the following:

1.   case_insensitive
2.  alias (TSV, TabSeperated), (NDJSON, JSONEache)
3.  "withName and withNameAndFormat"

Reply:
my design:

1.  to use factory only for name-> OutputFormatType
2.  OutputFormatType  is used as a unique id (no alias) to carry around(after sql parse) and ease ops like match, cmp, in..
3. one function from OutputFormatType to create dyn OutputFormat